rust/src/db.rs

```diff
@@ -323,6 +324,36 @@ pub async fn load_payments() -> Result<Vec<PaymentInfo>> {
     Ok(payments)
 }
 
+/// Update status of an expired lightning payments,
+/// as LDK does not handle monitoring for expired ones.
+pub async fn clean_expired_payments() -> Result<()> {
+    let newly_expired = load_payments()
+        .await?
+        .iter()
+        // only act on pending payments
+        .filter(|info| info.status == HTLCStatus::Pending)
+        // check whether we have reached expiry time
+        .filter(|pending| {
+            pending
+                .expiry_timestamp
+                .and_then(|expiry| {
+                    if get_timestamp() > expiry {
+                        Some(expiry)
+                    } else {
+                        None
+                    }
+                })
+                .is_some()
+        })
+        .cloned()
+        .collect::<Vec<PaymentInfo>>();
+
+    for payment in newly_expired {
+        update_payment(&payment.hash, HTLCStatus::Failed, None).await?;
+    }
```

Comment:
🙃 Using `iter_mut().for_each()` might be a bit simpler.

Reply:
I don't mind re-working it, I believe it's exactly the same under-the-hood for this syntactic sugar :)

Reply:
I don't think I can use `for_each` here, as async closures are still unstable....

Reply:
Yep, I didn't notice the `await` below! Definitely better to go for a for loop here.